components/camel-apns/src/main/java/org/apache/camel/component/apns/ApnsConsumer.java

```diff
@@ -82,15 +82,15 @@ public ApnsEndpoint getEndpoint() {
     }
 
     @Override
-    protected void doStart() throws Exception {
+    protected void doInit() throws Exception {
         // only add as consumer if not already registered
         if (!getEndpoint().getConsumers().contains(this)) {
             if (!getEndpoint().getConsumers().isEmpty()) {
                 throw new IllegalStateException("Endpoint " + getEndpoint().getEndpointUri() + " only allows 1 active consumer but you attempted to start a 2nd consumer.");
             }
             getEndpoint().getConsumers().add(this);
```

Comment:
Same thing here as before, we should do this in doStart as its adding and removing consumers on the endpoint at runtime when its started/stopped.

Reply:
Fixed